object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/framework/LombokSupport.java

```diff
@@ -0,0 +1,203 @@
+  /** The list of annotations that Lombok treats as non-null. */
+  public static final List<String> NONNULL_ANNOTATIONS =
+      Collections.unmodifiableList(
+          Arrays.asList(
+              "android.annotation.NonNull",
+              "android.support.annotation.NonNull",
+              "com.sun.istack.internal.NotNull",
+              "edu.umd.cs.findbugs.annotations.NonNull",
+              "javax.annotation.Nonnull",
+              // "javax.validation.constraints.NotNull", // The field might contain a null value
+              // until it is persisted.
+              "lombok.NonNull",
+              "org.checkerframework.checker.nullness.qual.NonNull",
+              "org.eclipse.jdt.annotation.NonNull",
+              "org.eclipse.jgit.annotations.NonNull",
+              "org.jetbrains.annotations.NotNull",
+              "org.jmlspecs.annotation.NonNull",
+              "org.netbeans.api.annotations.common.NonNull",
+              "org.springframework.lang.NonNull"));
+
+  // Keep a cache of these so that when declarationFromElement doesn't work,
+  // we can still default correctly. Value is the property name to treat as
+  // defaulted.
+  private final Map<Element, String> defaultedElements = new HashMap<>();
+
+  /**
+   * For {@code build} methods on {@code Builder} types, the framework support should determine the
+   * required properties and add a corresponding {@link
+   * org.checkerframework.checker.objectconstruction.qual.CalledMethods} annotation to the receiver.
+   *
+   * @param t a method that is possibly the {@code build} method for a builder. The only guaranteed
+   *     condition is that the enclosing class for the method is itself an inner class.
+   */
+  @Override
+  public void handlePossibleBuilderBuildMethod(AnnotatedExecutableType t) {
+    ExecutableElement element = t.getElement();
+
+    TypeElement enclosingElement = (TypeElement) element.getEnclosingElement();
+    Element nextEnclosingElement = enclosingElement.getEnclosingElement();
+
+    if ((FrameworkSupportUtils.hasAnnotation(enclosingElement, "lombok.Generated")
+            || FrameworkSupportUtils.hasAnnotation(element, "lombok.Generated"))
+        && enclosingElement.getSimpleName().toString().endsWith("Builder")) {
+      if (isBuilderBuildMethod(element, nextEnclosingElement)) {
+        List<String> requiredProperties = getLombokRequiredProperties(nextEnclosingElement);
+        AnnotationMirror newCalledMethodsAnno =
+            atypeFactory.createCalledMethods(requiredProperties.toArray(new String[0]));
+        t.getReceiverType().addAnnotation(newCalledMethodsAnno);
+      }
+    }
+  }
+
+  /**
+   * Allows for supporting a framework's {@code toBuilder} routine. Typically, the returned Builder
+   * setters invoked. So, the framework support can add a {@link
+   * org.checkerframework.checker.objectconstruction.qual.CalledMethods} annotation capturing this
+   * fact.
+   *
+   * @param t method named "toBuilder"
```

Comment:
I thought this could be any method (especially given the name of the method is `handlePossibleToBuilder`, not `handleToBuilder`)

Reply:
@kelloggm we know it's a method _named_ `toBuilder`, but not that it's a `toBuilder` method for the corresponding framework.  I clarified the Javadoc in the `FrameworkSupport` interface and deleted the Javadoc in the two implementations since that was identical